packages/botonic-plugin-nlu/src/nlu.js

```diff
@@ -0,0 +1,154 @@
+}
+
+const isProd = () => {
+  return process.env.STATIC_URL !== undefined
+}
+
+global.fetch = fetch
+
+async function resolveEnv() {
+  if (isProd() || isBrowser()) {
+    if (isProd()) {
+      return {
+        mode: 'prod',
+        uri: `${process.env.STATIC_URL}/${ASSETS_DIRNAME}/${MODELS_DIRNAME}/`
+      }
+    } else {
+      return { mode: 'dev', uri: window.location.href }
+    }
+  } else {
+    return {
+      mode: 'node',
+      uri: `${path.join(process.cwd(), NLU_PATH, MODELS_DIRNAME)}/`
+    }
+  }
+}
+
+function loadOption(lang, env) {
+  let nlu = {}
+  try {
+    if (env.mode === 'prod' || env.mode === 'dev') {
+      nlu.nluData = axios({
+        url: `${env.uri}${lang}/${NLU_DATA_FILENAME}`
+      })
+      nlu.model = tf.loadLayersModel(`${env.uri}${lang}/${MODEL_FILENAME}`)
+    } else {
+      nlu.nluData = JSON.parse(
+        readFile(`${env.uri}/${lang}/${NLU_DATA_FILENAME}`)
+      )
+      nlu.model = tf.loadLayersModel(
+        `file://${env.uri}${lang}/${MODEL_FILENAME}`
+      )
+    }
+  } catch (e) {
+    console.log('Cannot retrieve NLU Information', e)
+  }
+  return nlu
+}
+
+export class NLU {
+  constructor(options) {
+    return (async () => {
+      this.env = await resolveEnv()
+      this.langs = []
+      this.nlus = {}
+      for (let config of options) {
+        this.langs.push(config.LANG)
+        let { nluData, model } = loadOption(config.LANG, this.env)
+        this.nlus[config.LANG] = {
+          lang: config.LANG,
+          model: model,
+          nluData: nluData
+        }
+      }
+      await Promise.all([
+        ...Object.values(this.nlus).map(nlu => nlu.model),
+        ...Object.values(this.nlus).map(nlu => nlu.nluData)
+      ])
+      for (let [k, v] of Object.entries(this.nlus)) {
+        let nluData = await v.nluData
+        let { intentsDict, maxSeqLength, vocabulary } =
+          this.env.mode === 'node' ? nluData : nluData.data
+        let model = await v.model
+        let { lang } = v
+        this.nlus[k] = {
+          lang: lang,
+          model: model,
+          intents: intentsDict,
+          maxSeqLength: maxSeqLength,
+          tokenizer: new Tokenizer(vocabulary)
+        }
+      }
+      return this
+    })()
+  }
+
+  predict(userInput, nlu) {
+    let sequences = nlu.tokenizer.samplesToSequences(userInput)
+    let paddedSequences = []
+    let predictions = []
+    for (let sequence of sequences) {
+      paddedSequences.push(
+        padSequences([sequence], nlu.maxSeqLength).dataSync()
+      )
+    }
+    for (let paddedSequence of paddedSequences) {
+      predictions.push(
+        nlu.model.predict(tf.tensor([paddedSequence])).dataSync()
+      )
+    }
+    return predictions
+  }
+
+  getIntent(prediction, nlu) {
+    let results = {}
+    results.intents = []
+    prediction.map((prob, i) => {
+      let res = {}
+      res['intent'] = `${nlu.intents[i]}`
+      res['prob'] = prob
```

Comment:
Please also ensure thath prod has the same form as luis (https://docs.microsoft.com/en-us/azure/cognitive-services/luis/luis-quickstart-intents-only#get-intent-prediction-from-the-endpoint), ie an array of intent:string, score: number objects. The glitch here is that then we use "confidence" for the score of the main intent, and "score" of the list. 
@ericmarcos I suggest changing our luis plugin to make it consistent. I would use "score" everywhere to make it shorter and more general (since confidence may involve some semantics that some NLUs may not fulfill). What do you think?

Reply:
We agreed to use "confidence" in both places to keep compatibility with current projects using the "confidence" field of the top intent.

Reply:
renamed as intent, confidence, intents: [{intent:String, confidence:Number}]